blocks/faq/src/app.tsx

```diff
@@ -0,0 +1,469 @@
+          [field]: value,
+        },
+      },
+    });
+  };
+
+  const createFrequentlyAskedQuestionEntity = useCallback(async () => {
+    if (readonly) {
+      return;
+    }
+
+    const createEntityResponse = await graphModule.createEntity({
+      data: {
+        entityTypeId: frequentlyAskedQuestionType,
+        properties: {},
+      },
+    });
+
+    const createdEntityId =
+      createEntityResponse.data?.metadata.recordId.entityId;
+
+    if (createdEntityId) {
+      await graphModule.createEntity({
+        data: {
+          entityTypeId: hasFrequentlyAskedQuestion,
+          properties: {},
+          linkData: {
+            leftEntityId: entityId,
+            rightEntityId: createdEntityId,
+            leftToRightOrder:
+              (questionLinkedEntities[questionLinkedEntities.length - 1]
+                ?.linkEntity.linkData?.leftToRightOrder ?? 0) + 1,
+          },
+        },
+      });
+    }
+  }, [graphModule, questionLinkedEntities, entityId, readonly]);
+
+  const addQuestion = useCallback(async () => {
+    setQuestions([...questions, { id: uuid(), properties: {} }]);
+    await createFrequentlyAskedQuestionEntity();
+  }, [questions, createFrequentlyAskedQuestionEntity]);
+
+  const updateQuestionField = async (
+    index: number,
+    value: string,
+    field: QuestionOrAnswer,
+  ) => {
+    const questionEntity = questionEntities[index];
+
+    if (!questionEntity) {
+      return;
+    }
+
+    await graphModule.updateEntity({
+      data: {
+        entityId: questionEntity.metadata.recordId.entityId,
+        entityTypeId: frequentlyAskedQuestionType,
+        properties: {
+          ...questionEntity.properties,
+          [field]: value,
+        },
+      },
+    });
+  };
+
+  const removeQuestion = (index: number) => {
+    setQuestions(
+      questions.map((question, questionIndex) => {
+        if (index === questionIndex) {
+          return { ...question, animatingOut: true };
+        }
+        return question;
+      }),
+    );
+  };
+
+  useEffect(() => {
+    if (!questionEntities.length) {
+      void addQuestion();
+    }
+    // We only want to run this once when the block is initiated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const schema = useMemo(() => {
+    const questionsWithTitle = questionEntities.filter(
+      ({ properties: { [questionKey]: schemaQuestion } }) => !!schemaQuestion,
+    );
+
+    return JSON.stringify({
+      "@context": "http://schema.org",
+      "@type": "FAQPage",
+      mainEntity: questionsWithTitle.map(
+        ({
+          properties: {
+            [questionKey]: schemaQuestion,
+            [answerKey]: schemaAnswer,
+          },
+        }) => ({
+          "@type": "Question",
+          name: schemaQuestion,
+          acceptedAnswer: {
+            "@type": "Answer",
+            text: schemaAnswer ?? schemaQuestion,
+          },
+        }),
+      ),
+    });
+  }, [questionEntities]);
```

Comment:
You probably already did this, but asking to make sure. Is this data tested using a tool like https://developers.google.com/search/docs/appearance/structured-data to make sure it's correct?

Reply:
Yeah I've tested this before but for reference (in portuguese, sorry!):
![image](https://user-images.githubusercontent.com/37453800/228884846-0cf15cd4-513e-4d3e-a529-2a64aca530e4.png)